internal/adapter/lsp/document.go

```diff
@@ -229,10 +246,47 @@ func (d *document) DocumentLinks() ([]documentLink, error) {
 			})
 		}
 
+		if insideFenced {
```

Comment:
Does it make sense to place this block above the declarations of `appendLink()`? Just to shave down some execution time? As otherwise, within a code block for example, `appendLink()` is being redeclared with data that will never be used, as that line will be fully skipped anyway, meaning that `appendLink()` won't even be called in its current declared state.